src/path_converters.h

```diff
@@ -339,74 +347,69 @@ class PathClipper
         if (m_do_clipping) {
             /* This is the slow path where we actually do clipping */
 
-            if (m_end_poly) {
-                m_end_poly = false;
-                return (agg::path_cmd_end_poly | agg::path_flags_close);
-            }
-
-            if (m_has_next) {
-                m_has_next = false;
-                *x = m_nextX;
-                *y = m_nextY;
-                return agg::path_cmd_line_to;
+            if (queue_pop(&code, x, y)) {
+                return code;
             }
 
             while ((code = m_source->vertex(x, y)) != agg::path_cmd_stop) {
-                if (code == (agg::path_cmd_end_poly | agg::path_flags_close)) {
+                switch (code) {
+                case (agg::path_cmd_end_poly | agg::path_flags_close):
```

Comment:
The value `agg::path_cmd_end_poly | agg::path_flags_close` is how Agg indicates the end of a path that is also closed.  Why that's not a single enum value, I don't know, but that's how it's done in Agg so we must follow.  If it makes the code clearer I could create a new alias:

```
end_and_close_poly = agg::path_cmd_end_poly | agg::path_flags_close
```